### PR TITLE
Add Smartparens support

### PR DESCRIPTION
**Pull request by rye**
For now, `smartparens-global-mode` is enabled and we are using the vanilla configuration provided by `smartparens`.  I'm open to feedback to make this work better for people and also to add new features, but bear in mind that these things can be added later on as we see the need for them.

### CODE REVIEW

**Review by cg505 — DISMISSED** [review]
Add tests please (just test that when you load the file smartparens-global-mode gets activated).

Also, we should go through any languages that dot has preliminary support for and make sure smartparens is working okay. Seem pretty good out of the box to me but we still ought to check. This might not be part of this PR or #55 but just wanted to throw it on your radar.

**Comment by rye** [comment]
>Add tests please (just test that when you load the file `smartparens-global-mode` gets activated).

This test will be a good first exposure to tests—I'll take a stab at it and request review again.

>Also, we should go through any languages that dot has preliminary support for and make sure smartparens is working okay. Seem pretty good out of the box to me but we still ought to check. This might not be part of this PR or #55 but just wanted to throw it on your radar.

For what it's worth, this has been tested fairly extensively (~15 minutes of checkride editing) with Ruby, HTML, JS, and CSS within `web-mode`, and of course, EmacsLisp.  Feel free to checkout and verify that this is the case for you, and if there's any weird stuff we can toss that in this first-pass configuration.  This is just the first step for this feature but will accomplish >90% of what we want to do.

Smartparens is freakishly extensible so if we want to add a custom tweak to this later, I'm all for it.

**Comment by rye** [comment]
The one caveat that I noticed was that if you delete a line containing a paren, the matching paren does not get deleted.  Smartparens, of course, has a solution for this:
>You may want to try `smartparens-strict-mode`. This enforces that pairs are always balanced, so commands like `kill-line` keep your code well-formed.

(From the [Smartparens README](https://github.com/Fuco1/smartparens/blob/f493fdba6af036497e0f0a204c7f7388888bf6b3/README.md#getting-started).)

**Comment by cg505** [comment]
I enable strict mode personally but it is VERY confusing if you don't understand what is going on and should NOT be enabled by default

**Comment by rye** [comment]
I agree—it's a bit overbearing and can cause problems if you are editing in a weird way.  It might be something that we can document or write an enabler for, but besides that it's probably best if people just turn it on themselves.

File: .emacs.d/lisp/package/dependencies.el [hunk 1]
```diff
@@ -20,6 +20,7 @@
     elixir-mode ;; for editing Elixir code
     rust-mode ;; for editing Rust code
     buttercup ;; for tests
+    smartparens ;; for dealing with paired control flow symbols
```
Comment (cg505) [inline .emacs.d/lisp/package/dependencies.el:23]: "paired control flow symbols" ok mr scientist lmao

**Review by cg505 — APPROVED** [review]
not really much to complain about here, looks good

File: .emacs.d/lisp/package/dependencies.el (hunk 1 above)
Comment (rye) [reply .emacs.d/lisp/package/dependencies.el:23]: lmao

**Pull request merged**